flink-ml-benchmark/src/test/java/org/apache/flink/ml/benchmark/BenchmarkTest.java

```diff
@@ -54,6 +56,33 @@ public void testParseJsonFile() throws Exception {
         assertTrue(benchmarks.containsKey("KMeansModel-1"));
     }
 
+    @Test
+    public void verifyJsonFileLegality() throws IOException, ClassNotFoundException {
+        File resourcesDir =
+                new File(
+                                this.getClass()
+                                        .getClassLoader()
+                                        .getResource("benchmark-demo.json")
+                                        .getPath())
+                        .getParentFile();
+        File[] jsonFiles =
+                resourcesDir.listFiles(
+                        (dir, name) ->
+                                name.endsWith(".json") && !name.equals("benchmark-demo.json"));
+
+        for (File file : jsonFiles) {
+            Map<String, Map<String, Map<String, ?>>> benchmarks =
+                    BenchmarkUtils.parseJsonFile(file.getAbsolutePath());
+            for (Map<String, Map<String, ?>> params : benchmarks.values()) {
+                ReadWriteUtils.instantiateWithParams(params.get("stage"));
```

Comment:
nit: we can also add verifications to these aspects:

- `jsonFiles` is not empty.
- `params` contains no other keys except `stage`, `inputData` and `modelData`.
- The `stage` object is a subclass of `Stage`.
- The `inputData` or `modelData` object is a subclass of `DataGenerator`.

Reply:
For the first aspect,`BenchmarkUtils.parseJsonFile` would check if `jsonfile` is empty. Others are accepted.